django_snowflake/compiler.py

```diff
@@ -0,0 +1,105 @@
+from itertools import chain
+
+from django.db.models import JSONField
+from django.db.models.sql import compiler
+
+
+class SQLInsertCompiler(compiler.SQLInsertCompiler):
+    def as_sql(self):
+        # We don't need quote_name_unless_alias() here, since these are all
+        # going to be column names (so we can avoid the extra overhead).
+        qn = self.connection.ops.quote_name
+        opts = self.query.get_meta()
+        insert_statement = self.connection.ops.insert_statement(
+            on_conflict=self.query.on_conflict,
+        )
+        result = ["%s %s" % (insert_statement, qn(opts.db_table))]
+        fields = self.query.fields or [opts.pk]
+        result.append("(%s)" % ", ".join(qn(f.column) for f in fields))
+
+        select_columns = []
+        if self.query.fields:
+            value_rows = [
+                [
+                    self.prepare_value(field, self.pre_save_val(field, obj))
+                    for field in fields
+                ]
+                for obj in self.query.objs
+            ]
+            has_json_field = False
+            for i, field in enumerate(fields, 1):
+                if isinstance(field, JSONField):
+                    has_json_field = True
+                    select_columns.append(f'parse_json(column{i})')
+                else:
+                    select_columns.append(f'column{i}')
```

Comment:
I wonder what the prefix `column` means here? In my opinion, the SQL query should look like below:
```
INSERT INTO
    MY_TAB
SELECT 
    $1, 
    parse_json($1) 
FROM 
    (
        VALUES('1', '[1,2,3]')
    ) 
```